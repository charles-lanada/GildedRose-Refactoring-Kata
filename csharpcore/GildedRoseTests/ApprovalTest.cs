using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using GildedRoseKata;
using Xunit;

namespace GildedRoseTests
{
    [UseReporter(typeof(RiderReporter))]
    public class ApprovalTest
    {
        [Fact]
        public void ThirtyDays()
        {
            var fakeoutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeoutput));
            Console.SetIn(new StringReader("a\n"));

            Program.Main(new string[] { });
            var output = fakeoutput.ToString();

            Approvals.Verify(output);
        }

        [Fact]
        public void GildedRoseTest()
        {
            var Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            Items.Add(new Item { Name = "bar", SellIn = 1, Quality = 2 });
            var app = new GildedRose(Items);
            app.UpdateQuality();

            var testString = string.Join(Environment.NewLine, app.Items);
            Approvals.Verify(testString);
        }
    }
}