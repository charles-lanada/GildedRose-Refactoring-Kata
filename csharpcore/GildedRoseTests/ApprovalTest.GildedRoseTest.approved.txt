[Aged Brie,-1,0] => Name: Aged Brie, SellIn: -2, Quality: 2
[Aged Brie,-1,49] => Name: Aged Brie, SellIn: -2, Quality: 50
[Aged Brie,-1,50] => Name: Aged Brie, SellIn: -2, Quality: 50
[Aged Brie,-1,1] => Name: Aged Brie, SellIn: -2, Quality: 3
[Aged Brie,-1,-1] => Name: Aged Brie, SellIn: -2, Quality: 1
[Aged Brie,0,0] => Name: Aged Brie, SellIn: -1, Quality: 2
[Aged Brie,0,49] => Name: Aged Brie, SellIn: -1, Quality: 50
[Aged Brie,0,50] => Name: Aged Brie, SellIn: -1, Quality: 50
[Aged Brie,0,1] => Name: Aged Brie, SellIn: -1, Quality: 3
[Aged Brie,0,-1] => Name: Aged Brie, SellIn: -1, Quality: 1
[Aged Brie,11,0] => Name: Aged Brie, SellIn: 10, Quality: 1
[Aged Brie,11,49] => Name: Aged Brie, SellIn: 10, Quality: 50
[Aged Brie,11,50] => Name: Aged Brie, SellIn: 10, Quality: 50
[Aged Brie,11,1] => Name: Aged Brie, SellIn: 10, Quality: 2
[Aged Brie,11,-1] => Name: Aged Brie, SellIn: 10, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,-1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,50] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,-1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,11,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 1
[Backstage passes to a TAFKAL80ETC concert,11,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,11,50] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,11,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 2
[Backstage passes to a TAFKAL80ETC concert,11,-1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 0
[Sulfuras, Hand of Ragnaros,-1,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 0
[Sulfuras, Hand of Ragnaros,-1,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 49
[Sulfuras, Hand of Ragnaros,-1,50] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 50
[Sulfuras, Hand of Ragnaros,-1,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 1
[Sulfuras, Hand of Ragnaros,-1,-1] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: -1
[Sulfuras, Hand of Ragnaros,0,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 0
[Sulfuras, Hand of Ragnaros,0,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 49
[Sulfuras, Hand of Ragnaros,0,50] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 50
[Sulfuras, Hand of Ragnaros,0,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 1
[Sulfuras, Hand of Ragnaros,0,-1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: -1
[Sulfuras, Hand of Ragnaros,11,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 0
[Sulfuras, Hand of Ragnaros,11,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 49
[Sulfuras, Hand of Ragnaros,11,50] => Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 50
[Sulfuras, Hand of Ragnaros,11,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 1
[Sulfuras, Hand of Ragnaros,11,-1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: -1
[unknown,-1,0] => Name: unknown, SellIn: -2, Quality: 0
[unknown,-1,49] => Name: unknown, SellIn: -2, Quality: 47
[unknown,-1,50] => Name: unknown, SellIn: -2, Quality: 48
[unknown,-1,1] => Name: unknown, SellIn: -2, Quality: 0
[unknown,-1,-1] => Name: unknown, SellIn: -2, Quality: -1
[unknown,0,0] => Name: unknown, SellIn: -1, Quality: 0
[unknown,0,49] => Name: unknown, SellIn: -1, Quality: 47
[unknown,0,50] => Name: unknown, SellIn: -1, Quality: 48
[unknown,0,1] => Name: unknown, SellIn: -1, Quality: 0
[unknown,0,-1] => Name: unknown, SellIn: -1, Quality: -1
[unknown,11,0] => Name: unknown, SellIn: 10, Quality: 0
[unknown,11,49] => Name: unknown, SellIn: 10, Quality: 48
[unknown,11,50] => Name: unknown, SellIn: 10, Quality: 49
[unknown,11,1] => Name: unknown, SellIn: 10, Quality: 0
[unknown,11,-1] => Name: unknown, SellIn: 10, Quality: -1
